src/scene_battle_rpg2k3.cpp

```diff
@@ -2268,32 +2313,34 @@ Scene_Battle_Rpg2k3::BattleActionReturn Scene_Battle_Rpg2k3::ProcessBattleAction
 
 Scene_Battle_Rpg2k3::BattleActionReturn Scene_Battle_Rpg2k3::ProcessBattleActionCBARangedWeaponMove(Game_BattleAlgorithm::AlgorithmBase* action) {
 	auto* source = action->GetSource();
-	Game_Battler* target = nullptr;
-	// The ranged weapon animation targets the original single target
-	// if the weapon has the "Attack All" flag set and the ranged attack
-	// range is set to "Single Enemy"
-	if (action->GetWeaponData()->attack_all) {
-		target = action->GetOriginalSingleTarget();
-	} else {
-		target = action->GetTarget();
-	}
 
 	if (cba_ranged_weapon_move_frame < cba_num_ranged_weapon_move_frames) {
 		cba_ranged_weapon_move_frame++;
-		int offset_x = 0;
-		int offset_y = 0;
-		if (target) {
-			offset_x = target->GetBattlePosition().x - source->GetBattlePosition().x;
-			offset_y = target->GetBattlePosition().y - source->GetBattlePosition().y;
+		for (auto& it: cba_ranged) {
+			int offset_x = 0;
+			int offset_y = 0;
+			if (action->GetWeaponData()->ranged_target == lcf::rpg::Item::Target_center) {
```

Comment:
About the bug: My guess is that the problem is here. This must also check if you attack all enemies.

Reply:
Yes, this is true. I'm 99% sure that this is the cause.

Reply:
That was indeed the cause. It is fixed now.